### ISSUE TITLE
GRPO: a NaN training step is invisible in log_history, so a corrupted run reports a clean loss curve

### ISSUE THREAD

**Issue by behroozazarkhalili** [huggingface/trl#6702]
### Summary

A training step whose loss is NaN is not visible in `log_history`. The reported loss curve stays finite and plausible while the step that produced it was NaN, so a run that is silently corrupting its weights looks healthy in the logs and in any tracker fed from them.

This came out of the regression test for #6166 (PR #6693). @albertvillanova asked for it to be tracked separately, since it is orthogonal to that fix.

### Mechanism

`transformers.Trainer` filters NaN and Inf out of the logged running loss by design:

```python
# transformers/trainer.py:1746
if (
    self.args.logging_nan_inf_filter
    and not is_torch_xla_available()
    and (torch.isnan(tr_loss_step) or torch.isinf(tr_loss_step))
):
    # if loss is nan or inf simply add the average of previous logged losses
    self._tr_loss += self._tr_loss / (1 + self.state.global_step - self._globalstep_last_logged)
else:
    ...
```

`logging_nan_inf_filter` defaults to `True` (`transformers/training_args.py:977`), and its own help text says it "Does not affect gradient computation, only logging". So the NaN step still runs `backward()` and still takes an optimizer step. Only the *report* is repaired, by substituting the mean of the previously logged losses.

### Why this matters more for GRPO than for plain SFT

The GRPO loss has several paths that can produce a NaN from data rather than from divergence: an unscorable token logprob coming back from vLLM (#6166), a degenerate reward group, an overflowing `exp` of a KL log-ratio (#3015). In those cases the NaN is not a symptom of a bad learning rate that a user would go looking for. It is a single poisoned token or group, and the resulting NaN gradients propagate into the weights while the loss curve shows nothing at all.

### Reproduction

Observed while building the #6166 test. With the fix removed, an individual step's loss is NaN at the point `_compute_loss` returns it, and `state.log_history["train_loss"]` is still finite for that same run. That is why the test in PR #6693 asserts on the losses returned by `_compute_loss` rather than on `log_history`: asserting on `log_history` passes even with the fix removed, which makes it a vacuous test.

### Possible directions

Listing these as options rather than a proposal, since the right layer is a maintainer call:

1. Count NaN and Inf steps and surface the count as a metric, so the filtering stays but stops being silent.
2. Emit a warning the first time a step is filtered, naming the step index.
3. Leave `transformers` alone and check for a non-finite loss inside the GRPO loss path, where the cause is known and can be named.

Option 3 keeps the change inside TRL and can point at the actual source, which the generic filter cannot do.

I am happy to implement whichever direction you prefer.

cc @qgallouedec @kashif

**Comment by behroozazarkhalili** [comment]
Closing this along with #6862. Nobody has hit it in a real training run. Happy to reopen if that changes.
